app/models/author.rb

```diff
@@ -35,6 +37,43 @@ def leap_year?
     LeapYear.new(created_at).leap_year?
   end
 
+  def zodiac
+    zodiacs = {
+      "Aries ♈️" => [Date.parse("March 21 #{ date_of_birth.year }"), Date.parse("April 19 #{ date_of_birth.year }")],
+      "Taurus ♉️" => [Date.parse("April 20 #{ date_of_birth.year }"), Date.parse("May 20 #{ date_of_birth.year }")],
+      "Gemini ♊️" => [Date.parse("May 21 #{ date_of_birth.year }"), Date.parse("June 20 #{ date_of_birth.year }")],
+      "Cancer ♋️" => [Date.parse("June 21 #{ date_of_birth.year }"), Date.parse("July 22 #{ date_of_birth.year }")],
+      "Leo ♌️" => [Date.parse("July 23 #{ date_of_birth.year }"), Date.parse("August 22 #{ date_of_birth.year }")],
+      "Virgo ♍️" => [Date.parse("August 23 #{ date_of_birth.year }"), Date.parse("September 22 #{ date_of_birth.year }")],
+      "Libra ♎️" => [Date.parse("September 23 #{ date_of_birth.year }"), Date.parse("October 22 #{ date_of_birth.year }")],
+      "Scorpio ♏️" => [Date.parse("October 23 #{ date_of_birth.year }"), Date.parse("November 21 #{ date_of_birth.year }")],
+      "Sagittarius ♐️" => [Date.parse("November 22 #{ date_of_birth.year }"), Date.parse("December 21 #{ date_of_birth.year }")],
+      # "Capricorn ♑️" => [Date.parse("December 22 #{ date_of_birth.year }"), Date.parse("January 19 #{ date_of_birth.year }")],
+      "Aquarius ♒️" => [Date.parse("January 20 #{ date_of_birth.year }"), Date.parse("February 18 #{ date_of_birth.year }")],
+      "Pisces ♓️" => [Date.parse("February 19 #{ date_of_birth.year }"), Date.parse("March 20 #{ date_of_birth.year }")],
+    }
```

Comment:
There's a lot of repetition in this block, with plenty of what the reader is ingesting being boilerplate code (EG: Date.parse, date_of_birth.year).

We should aim to reduce the amount of boilerplate code we need to read, so that we can focus on what matters.

In this case, we really want to focus primarily on the day and months.

We could use a lambda to achieve this:

```ruby
date_for = -> (month_and_day) { Date.parse("#{ month_and_day } #{ date_of_birth.year }") }

zodiacs = {
  "Aries ♈️" => [date_for.call("March 21"), date_for.call("April 19")],
  "Taurus ♉️" => [date_for.call("April 20"), date_for.call("May 20")],
  "Gemini ♊️" => [date_for.call("May 21"), date_for.call("June 20")],
  "Cancer ♋️" => [date_for.call("June 21"), date_for.call("July 22")],
  "Leo ♌️" => [date_for.call("July 23"), date_for.call("August 22")],
  "Virgo ♍️" => [date_for.call("August 23"), date_for.call("September 22")],
  "Libra ♎️" => [date_for.call("September 23"), date_for.call("October 22")],
  "Scorpio ♏️" => [date_for.call("October 23"), date_for.call("November 21")],
  "Sagittarius ♐️" => [date_for.call("November 22"), date_for.call("December 21")],
  # "Capricorn ♑️" => [date_for.call("December 22"), date_for.call("January 19")],
  "Aquarius ♒️" => [date_for.call("January 20"), date_for.call("February 18")],
  "Pisces ♓️" => [date_for.call("February 19"), date_for.call("March 20")],
}
```

Or, we could take another path entirely, and move the date parsing somewhere else:

```ruby
zodiacs = {
  "Aries ♈️" => ["March 21", "April 19"],
  "Taurus ♉️" => ["April 20", "May 20"],
  "Gemini ♊️" => ["May 21", "June 20"],
  "Cancer ♋️" => ["June 21", "July 22"],
  "Leo ♌️" => ["July 23", "August 22"],
  "Virgo ♍️" => ["August 23", "September 22"],
  "Libra ♎️" => ["September 23", "October 22"],
  "Scorpio ♏️" => ["October 23", "November 21"],
  "Sagittarius ♐️" => ["November 22", "December 21"],
  # "Capricorn ♑️" => ["December 22", "January 19"],
  "Aquarius ♒️" => ["January 20", "February 18"],
  "Pisces ♓️" => ["February 19", "March 20"],
}

zodiac_array = zodiacs.find do |zodiac, (begin_month_and_day, end_month_and_day)|
  begin_date = Date.parse("#{ begin_month_and_day } #{ date_of_birth.year }")
  end_date = Date.parse("#{ end_month_and_day } #{ date_of_birth.year }")

  date_of_birth.between?(begin_date, end_date)
end
```

Reply:
I prefer the second option! It seems more intuitive to me. I'm going to implement this.